iphone/Classes/UtilsModule.m

```diff
@@ -58,7 +58,17 @@ - (TiBlob *)base64decode:(id)args
 {
   ENSURE_SINGLE_ARG(args, NSObject);
 
-  NSString *str = [[self convertToString:args] stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceAndNewlineCharacterSet]];
+  NSMutableString *str = [[NSMutableString alloc] initWithString:[[self convertToString:args]
+                                                                     stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceAndNewlineCharacterSet]]];
+  // Fill padding for backwards compatibility
+  while ([str length] % 4 != 0) {
```

Comment:
I don't think this is right. Base64 encoded strings will only have one or two '=' characters as padding, which suggest modulo 3, not 4.

https://stackoverflow.com/questions/4080988/why-does-base64-encoding-require-padding-if-the-input-length-is-not-divisible-by

Reply:
The RFC sounds helpful:

```
(1) the final quantum of encoding input is an integral multiple of 24
   bits; here, the final unit of encoded output will be an integral
   multiple of 4 characters with no "=" padding,

   (2) the final quantum of encoding input is exactly 8 bits; here, the
   final unit of encoded output will be two characters followed by two
   "=" padding characters, or

   (3) the final quantum of encoding input is exactly 16 bits; here, the
   final unit of encoded output will be three characters followed by one
   "=" padding character.
```